Assets/SEE/DataModel/DG/MergeDiffGraphExtension.cs

```diff
@@ -0,0 +1,260 @@
+﻿using System.Collections.Generic;
+using System;
+using Sirenix.Utilities;
+
+namespace SEE.DataModel.DG
+{
+    /// <summary>
+    /// Merges the differences between an old and a new graph into the new graph.
+    /// </summary>
+    internal static class MergeDiffGraphExtension
+    {
+        /// <summary>
+        /// This postfix will be added at the end of the name of an attribute whose
+        /// value has changed or was deleted from the old graph to the new graph.
+        /// </summary>
+        public const string AttributeOldValuePostfix = ".Old";
+
+        /// <summary>
+        /// Merges the changes of nodes and edges of <paramref name="newGraph"/> with
+        /// respect to <paramref name="oldGraph"/>.
+        ///
+        /// More precisely, graph elements in <paramref name="newGraph"/>, but not in
+        /// <paramref name="oldGraph"/> are marked by toggle attribute
+        /// <see cref="ChangeMarkers.IsNew"/>.
+        ///
+        /// Graph elements in <paramref name="oldGraph"/>, but not in <paramref name="newGraph"/>
+        /// are added to <paramref name="newGraph"/> and marked by toggle attribute
+        /// <see cref="ChangeMarkers.IsDeleted"/>.
+        ///
+        /// If a graph element is in both <paramref name="newGraph"/> and
+        /// <paramref name="oldGraph"/> and has a difference in any of its attributes,
+        /// the graph element in <paramref name="newGraph"/> will be marked by toggle
+        /// attribute <see cref="ChangeMarkers.IsChanged"/>. Let N be the graph
+        /// element in <paramref name="newGraph"/> corresponding to the graph element O
+        /// of <paramref name="oldGraph"/>. For every attribute, A of O,
+        /// either not present in N or present in N but with a different value, an attribute
+        /// named A appended by the <see cref="AttributeOldValuePostfix"/>
+        /// will be added to N with the value of O.A.
+        ///
+        /// Whether two attribute values are the equal/different is determined
+        /// by <see cref="Object.Equals(object, object)"/>.
+        ///
+        /// Whether two graph elements are the same in the two graphs is determined by
+        /// <see cref="GraphElement.ID"/>.
+        ///
+        /// Only <paramref name="newGraph"/> can be effected by this method.
```

Comment:
Here opinions differ. I learnt from the book "Bugs in writing" that _effected_ is actually the right term because _affected_ relates to feelings. But there are also other debates as, for instance, [here](https://www.grammarly.com/blog/affect-vs-effect/). But even in that debate, which appears to favor _affected_ over _effected_, ("Use affect as the verb in a sentence when you’re talking about producing change or making a difference. For example, a new discovery can affect a scientific theory, and failing a test can affect someone’s mood.") the authors write:  "Effect as a verb means to directly achieve or bring something about. Affect as a less common verb means to put on a pretense."

It's a bit inconclusive.